ecosystem/typescript/sdk/src/providers/indexer.ts

```diff
@@ -0,0 +1,81 @@
+import axios from "axios";
+
+import { AnyNumber } from "../bcs/types";
+import { MaybeHexString } from "../hex_string";
+import { GetAccountCurrentTokensQuery, GetTokenActivitiesQuery } from "../indexer/generated/operations";
+import { GetAccountCurrentTokens, GetTokenActivities } from "../indexer/generated/queries";
+
+interface PaginationArgs {
+  offset?: AnyNumber;
+  limit?: number;
+}
+
+type GraphqlQuery = {
+  query: string;
+  variables?: {};
+};
+/**
+ * Provides methods for retrieving data from Aptos Indexer.
+ * For more detailed Queries specification see
+ * {@link https://cloud.hasura.io/public/graphiql?endpoint=https://indexer.mainnet.aptoslabs.com/v1/graphql}
+ */
+export class IndexerClient {
+  endpoint: string;
+
+  /**
+   * @param endpoint URL of the Aptos Indexer API endpoint.
+   */
+  constructor(endpoint: string) {
+    this.endpoint = endpoint;
+  }
+
+  /**
+   * Builds a axios client call to fetch data from Aptos Indexer.
+   *
+   * @param graphqlQuery A GraphQL query to pass in the `data` axios call.
+   */
+  private async queryIndexer(graphqlQuery: GraphqlQuery): Promise<any> {
+    try {
+      const { data } = await axios({
+        url: this.endpoint,
+        method: "post",
+        data: graphqlQuery,
+      });
+      if (data.errors) {
+        return data.errors;
```

Comment:
should you wrap the error in an exception and throw it? I know graphql sometimes returns partial errors. e.g. you can get partial data back along with some errors. How do you want to handle it in this case?

Reply:
yes, good point on wrapping it. 
As for partial error data, I think it returns as an error within the query response and not as a full/status error, something like. @bowenyang007 thoughts?
```
current_token_ownerships {
  last_transaction_version
  ...
  current_token_data {
    error {
      message
    }
    creator_address
  }
}
```